### PR TITLE
sfdx RNs: Nov 3

### PR DESCRIPTION
**Pull request by jshackell-sfdc**

### CODE REVIEW

File: releasenotes/sfdx/README.md [hunk 1]
```diff
@@ -21,10 +21,28 @@ Additional documentation:
...
+
+    Use the `--command` parameter to run a specific command in debug mode; the doctor writes both stdout and stderr to separate `*.log` files. Encapsulate the command in double quotes. For example:
+
+    `sfdx doctor --command "force:org:list --all"`
+
+    To run diagnostic tests on a specific plugin rather than the CLI itself, use the `--plugin` parameter. If the plugin isn't listening to the doctor, then you get a warning and no data.
+
+    `sfdx doctor --plugin @salesforce/plugin-source`
+
+    The CLI doctor is in and ready to diagnose all your problems!
```
Comment (shetzel) [inline releasenotes/sfdx/README.md:44]: Same comment as for `sf doctor` - add the `--createissue` flag

**Review by shetzel — CHANGES_REQUESTED** [review]

File: releasenotes/sfdx/README.md (hunk 1 above)
Comment (jshackell-sfdc) [reply releasenotes/sfdx/README.md:44]: done!  thx.

File: releasenotes/sfdx/README.md [hunk 2]
```diff
@@ -21,10 +21,99 @@ Additional documentation:
...
+
+    To run diagnostic tests on a specific plugin rather than the CLI itself, use the `--plugin` parameter. If the plugin isn't listening to the doctor, then you get a warning and no data.
+
+    `sfdx doctor --plugin @salesforce/plugin-source`
+    
+    We've made it really easy to create a GitHub issue: specify the `--createissue` parameter, enter a title at the prompt, and a browser window automatically opens with a partially-filled GitHub issue. Enter the remaining information about your specific issue, click **Submit new issue**, and you're done.  Easy peasy!
+
+    The CLI doctor is in and ready to diagnose all your problems!
+    
+* NEW: You can now automatically replace snippets of your metadata source files with specific values right before you deploy the files to an org with the `force:source:deploy|push` commands. Use this new feature to, for example, replace the endpoint in a NamedCredential, depending on whether you're deploying to a production or scratch org. Or specify a password in an ExternalDataSource that you don't want to store in your repo. The use cases are endless!
```
Comment (mshanemc) [inline releasenotes/sfdx/README.md]: Let's say something about these not persisting to the filesystem ("ephemeral") or "during the deployment without writing them to your project", etc.

File: releasenotes/sfdx/README.md [hunk 3]
```diff
@@ -21,10 +21,99 @@ Additional documentation:
...
+          "filename": "force-app/main/default/classes/myClass.cls",
+          "stringToReplace": "replaceMe",
+          "replaceWithEnv": "THE_REPLACEMENT"  
+        }
+      ],
+    …
+    }
+    ```
+
+    You can specify these keys in the `replacements` property:
```
Comment (mshanemc) [inline releasenotes/sfdx/README.md:71]: I *think* we want to point out the 2 places people are likely to run into problems
1. always use forward slashes, even on windows
2. JSON requires that you escape any backslashes with another backslash

File: releasenotes/sfdx/README.md [hunk 4]
```diff
@@ -21,10 +21,99 @@ Additional documentation:
...
+    * `glob`: Collection of files that contain the string to be replaced. Example: `**/classes/*.cls`.
+    * `stringToReplace`: The string to be replaced.
+    * `regexToReplace`: Regular expression that specifies a string pattern to be replaced. 
+    * `replaceWithEnv`: Specifies that the string be replaced with the value of the environment variable.
+    * `replaceWithFile`: Specifies that the string be replaced with the contents of a file.
+    * `replaceWhenEnv`: Specifies a condition, using environment variables, for when a string replacement occurs. 
+
+    This example is similar to the previous one, except that the replacement occurs only if an environment variable called `DEPLOY_DESTINATION` exists and it has a value of `PROD`.
+
+    ```bash
```
Comment (mshanemc) [inline releasenotes/sfdx/README.md]:
```suggestion
    ```json 
```

File: releasenotes/sfdx/README.md [hunk 5]
```diff
@@ -21,10 +21,99 @@ Additional documentation:
...
+
+    The CLI doctor is in and ready to diagnose all your problems!
+    
+* NEW: You can now automatically replace snippets of your metadata source files with specific values right before you deploy the files to an org with the `force:source:deploy|push` commands. Use this new feature to, for example, replace the endpoint in a NamedCredential, depending on whether you're deploying to a production or scratch org. Or specify a password in an ExternalDataSource that you don't want to store in your repo. The use cases are endless!
+
+    To configure string replacement, add a `replacements` property to your `sfdx-project.json` file and use key-value pairs to describe how the string replacement works. 
+
+    For example, this `sfdx-project.json` snippet specifies that when you deploy the `force-app/main/default/classes/myClass.cls` source file, all occurrences of the string `replaceMe` are replaced with the value of the `THE_REPLACEMENT` environment variable:
+
+    ```bash
```
Comment (mshanemc) [inline releasenotes/sfdx/README.md]:
```suggestion
    ```json
```

File: releasenotes/sfdx/README.md [hunk 6]
```diff
@@ -21,10 +21,99 @@ Additional documentation:
...
+
+    This example shows how to retrieve all Apex classes from your default org and put the source-formatted files into the `unpackaged-files` directory. If this directory doesn't exist, the command creates it for you. Your configured package directories are unchanged.
+
+    `sfdx force:source:retrieve --retrievetargetdir ./unpackaged-files --metadata ApexClass`
+
+    Many thanks to [Matthias Rolke](https://github.com/amtrack) for suggesting the cool feature, and then writing a lot of the code!  [plugin-source PR #426](https://github.com/salesforcecli/plugin-source/pull/426)
+
+* CHANGE: We upgraded the version of Node.js contained in the full Salesforce CLI Docker image to LTS v18. (sfdx-cli PR [#720](https://github.com/salesforcecli/sfdx-cli/pull/720))
+
+* FIX: The `force:source:*` commands now support these metdata types:
```
Comment (mshanemc) [inline releasenotes/sfdx/README.md:114]: they tried to add these previously, and had the names slightly wrong.  I can't remember if those went out last release or not.  If not, they're net new and all good.

Comment (jshackell-sfdc) [reply releasenotes/sfdx/README.md:114]: yeah, I noticed that. And yeah, I doc'd 'em last release.  I'll figure out which names changed and list them that way. Thx.

File: releasenotes/sfdx/README.md (hunk 2 above)
Comment (jshackell-sfdc) [reply releasenotes/sfdx/README.md]: Nice

File: releasenotes/sfdx/README.md (hunk 3 above)
Comment (jshackell-sfdc) [reply releasenotes/sfdx/README.md:71]: Added

**Pull request merged**